modules/opensearch/includes/opensearch.search.inc

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * @file
+ * Implements handling of loans through OpenRuth.
+ */
+
+/**
+ * Get a list of material types from the Well.
+ */
+function opensearch_search_material_types() {
+  // Get a list of types by executing a null query and look at the facets
+  // result.
+  $options = [
+    'facets' => ['facet.type'],
+    'numFacets' => 99,
+    'reply_only' => TRUE,
+    'sort' => 'random',
+  ];
+  module_load_include('client.inc', 'ting');
+  $result = ting_do_search("*", 0, 0, $options);
+
+  $types = [];
+  foreach ($result->facets['facet.type']->terms as $term => $count) {
+    $types[drupal_strtolower($term)] = $count;
+  }
+
+  // Only save if we actually got any types.
+  if (!empty($types)) {
+    variable_set('ting_well_types', $types);
+  }
```

Comment:
As I see it this variable does not belong in the `opensearch` module.

In general relying on a shared variable is not a very pretty solution. I would prefer that external code went through either the provider or `_ting_fetch_well_types` but that is perhaps outside the scope of this PR.

As an initial step I suggest we move setting of the variable back to `_ting_fetch_well_types()`. This keeps the `opensearch` provider clean and avoids "namespace" mixup.